### DISCUSSION TITLE
[firebase_messaging] onBackgroundMessage - how to use multiple multi-language notifications

### DISCUSSION THREAD

**Discussion by GuyDev1** [firebase/flutterfire#3878] [Q&A]
Hey,

So I'm using Flutter_Localization in order to parse the app's texts in multiple languages.
When it comes to notifications, I could use it as well (by defining constants and handling the translation in app (using onMessage for example to parse it).

However, onBackgroundMessage requires passing a function which is outside the scope of the class / a static method.
Therefore I can't see how I could call the Localization instance (which requires a context) in order to translate the given data.

Any ideas would be much appreciated :)

**Comment by QuentinSc** [comment]
Hello @GuyDev1,

Did you succeeded ?

Thank you !

**Reply by GuyDev1** [reply]
Nope, didn't deal with localization since then lol.
